staging/src/k8s.io/client-go/tools/cache/the_real_fifo.go

```diff
@@ -158,6 +170,9 @@ func (f *RealFIFO) addToItems_locked(deltaActionType DeltaType, skipTransform bo
 		Type:   deltaActionType,
 		Object: obj,
 	})
+	if f.metrics != nil {
+		f.metrics.numberOfQueuedItem.Set(float64(len(f.items)))
```

Comment:
I historically remember some issue (this was long time ago though) of metrics causing some lock contention.
Given this being on a critical path and under lock - do we have any data showing that it will not have performance impact?